### PR TITLE
feat: add download link to versions PE-1135

### PR DESCRIPTION
**Pull request by jdaev**
Adds a download link to the specific versions in the activity sidebar.

### CODE REVIEW

File: lib/pages/drive_detail/components/fs_entry_side_sheet.dart [hunk 1]
```diff
@@ -367,8 +367,32 @@ class FsEntrySideSheet extends StatelessWidget {
                       } else if (revision is FileRevisionWithTransactions) {
                         switch (revision.action) {
                           case RevisionAction.create:
-                            content = Text(
-                                'This file was created with the name ${revision.name}.');
```
Comment (matibat) [inline lib/pages/drive_detail/components/fs_entry_side_sheet.dart:371]: UI internationalization: do we actually need to split this text?

File: lib/pages/drive_detail/components/fs_entry_side_sheet.dart [hunk 2]
```diff
@@ -378,8 +402,33 @@ class FsEntrySideSheet extends StatelessWidget {
                             content = Text('This file was moved.');
                             break;
                           case RevisionAction.uploadNewVersion:
-                            content = Text(
-                                'A new version of this file was uploaded.');
```
Comment (matibat) [inline lib/pages/drive_detail/components/fs_entry_side_sheet.dart:382]: This has to be split as well :thinking:

File: lib/pages/drive_detail/components/fs_entry_side_sheet.dart (hunk 1 above)
Comment (jdaev) [reply lib/pages/drive_detail/components/fs_entry_side_sheet.dart:371]: Yeah, it's the design that was on the ticket. Maybe we should revise it to make it easier for you.

**Review by thiagocarvalhodev — DISMISSED** [review]
LGTM

File: lib/blocs/file_download/personal_file_download_cubit.dart [hunk 3]
```diff
@@ -3,8 +3,9 @@ part of 'file_download_cubit.dart';
 /// [ProfileFileDownloadCubit] includes logic to allow a user to download files
 /// that they have attached to their profile.
 class ProfileFileDownloadCubit extends FileDownloadCubit {
-  final String driveId;
-  final String fileId;
+  final DriveID driveId;
+  final FileID fileId;
+  final String dataTxId;
```
Comment (karlprieb) [inline lib/blocs/file_download/personal_file_download_cubit.dart:8]: We could create a DataTxID type on `string_types.dart` for this one.

File: lib/components/file_download_dialog.dart [hunk 4]
```diff
@@ -15,15 +16,17 @@ import 'components.dart';
 
 Future<void> promptToDownloadProfileFile({
   required BuildContext context,
-  required String driveId,
-  required String fileId,
+  required DriveID driveId,
+  required FileID fileId,
+  required String dataTxId,
```
Comment (karlprieb) [inline lib/components/file_download_dialog.dart:21]: We could create a DataTxID type on `string_types.dart` for this one.

File: lib/blocs/file_download/personal_file_download_cubit.dart (hunk 3 above)
Comment (jdaev) [reply lib/blocs/file_download/personal_file_download_cubit.dart]: [Done](https://github.com/ardriveapp/ardrive-web/pull/424/commits/d6568978dca496069bf56cc9f81d3aeaf49a0f13)

**Review by thiagocarvalhodev — APPROVED** [review]

**Review by karlprieb — DISMISSED** [review]

File: lib/pages/drive_detail/components/fs_entry_side_sheet.dart (hunk 1 above)
Comment (matibat) [reply lib/pages/drive_detail/components/fs_entry_side_sheet.dart]: So we ended up re-designing it. Now there's no need to split it :+1:

**Review by matibat — APPROVED** [review]
LGTM!

**Pull request merged**